### PR TITLE
Update workloads base image

### PR DESCRIPTION
**Pull request by tunguyen9889**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -45,7 +45,8 @@ jobs:
           context: .
           file: ./Dockerfile
           # platforms: linux/amd64,linux/arm64,linux/386
-          push: ${{ github.event_name != 'pull_request' }}
+          # push: ${{ github.event_name != 'pull_request' }}
+          push: true
```
Comment (gyng) [inline .github/workflows/ci.yaml:49]: Is this for testing?

**Review by gyng — COMMENTED** [review]
:100:

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (tunguyen9889) [reply .github/workflows/ci.yaml:49]: yes, I temporary changed it to force push image.

File: Dockerfile [hunk 2]
```diff
@@ -13,9 +13,13 @@ ENV PATH ${PATH}:${SPARK_HOME}/bin
 
 WORKDIR ${SPARK_HOME}
 
-RUN apt-get update && apt-get install -y \
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    gcc \
+    build-essential \
```
Comment (sweatybridge) [inline Dockerfile:19]: Run succeeded: https://amoy.ai/project/view/qiao/training/pipeline/view/churn-aws/run/10/step/preprocess/logs

```suggestion
    build-essential \
```

`gcc` is included in `build-essential`.

**Review by sweatybridge — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -58,22 +58,22 @@
...
+			<version>3.3.1</version>
 		</dependency>
 
 		<!-- https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-bundle -->
 		<!-- >= 1.11.704 needed for IRSA; hadoop-aws 3.2.0's version is too old -->
 		<dependency>
 			<groupId>com.amazonaws</groupId>
 			<artifactId>aws-java-sdk-bundle</artifactId>
-			<version>1.12.5</version>
+			<version>1.12.129</version>
```
Comment (sweatybridge) [inline pom.xml:69]: Since we've upgraded to hadoop-aws 3.3.1, do we still need this separate dependency?

Comment (sweatybridge) [reply pom.xml:69]: seems like we need it https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws/3.3.1

**Review by sweatybridge — CHANGES_REQUESTED** [review]
let me test on gcp before approving

File: pom.xml [hunk 4]
```diff
@@ -36,7 +36,7 @@
 		<dependency>
 			<groupId>com.google.cloud.bigdataoss</groupId>
 			<artifactId>gcs-connector</artifactId>
-			<version>hadoop3-1.9.18</version>
+			<version>hadoop3-2.2.4</version>
 			<!-- Can be removed after upgrading to hadoop 3.3.0 -->
```
Comment (sweatybridge) [inline pom.xml:40]:
```suggestion
			<!-- Can be removed after upgrading to hadoop 3.3.0 -->
```
we can remove these dependencies after upgrading to spark 3.2

**Review by sweatybridge — APPROVED** [review]

**Pull request merged**